extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -194,7 +195,7 @@ public boolean start()
             public void run()
             {
               while (!executorService.isShutdown()) {
-                final ConsumerConnector consumerConnector = buildConnector(kafkaProperties);
+                consumerConnector = buildConnector(kafkaProperties);
```

Comment:
I think the race is still there:

1. kafka lookup thingy is running happily for a while
2. executor thread's consumer crashes, it re-loops to the top of the `while` 
3. executor thread checks `executorService.isShutdown()`, finds it's false and continues into the body
4. close is called in some other thread
5. close thread calls `executorService.shutdown()`
6. close thread closes a `consumerConnector` from a previous loop of the executor
7. executor thread creates a new consumerConnector and does its thing, never exiting

I think this can be addressed by having close call `shutdownNow` instead of `shutdown`, and the executor thread exiting if interrupted (rather than re-looping) by (a) exiting on InterruptedException, (b) checking for interrupt flag on each message read.